apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java

```diff
@@ -271,47 +333,59 @@ private static void buildNPIResource(
-        BufferedWriter out = new BufferedWriter(new OutputStreamWriter(fw, outEnc))) {
-
-      // Get indexes according to header
-      String line = reader.readLine();
-      String[] fields = line.split(",");
-      Map<String, Integer> indexes = getIndexNumbers(fields);
-      Integer npiIndex = getIndexNumberForField(indexes, "NPI");
-      Integer entityTypeCodeIndex = getIndexNumberForField(indexes, "Entity Type Code");
-      Integer orgNameIndex =
-          getIndexNumberForField(indexes, "Provider Organization Name (Legal Business Name)");
-
-      while ((line = reader.readLine()) != null) {
-        fields = line.split(",");
-
-        String orgName = fields[orgNameIndex].trim().replace("\"", "");
-        String npi = fields[npiIndex].trim().replace("\"", "");
-        String entityTypeCode = fields[entityTypeCodeIndex].trim().replace("\"", "");
-
-        // entity type code 2 is organization
-        if (!Strings.isNullOrEmpty(entityTypeCode) && Integer.parseInt(entityTypeCode) == 2) {
-          out.write(npi + "\t" + orgName);
-          out.newLine();
-        }
+        DeflaterOutputStream dos = new DeflaterOutputStream(fw); ) {
+      CSVParser csvParser =
+          new CSVParser(
+              reader,
+              CSVFormat.DEFAULT.withFirstRecordAsHeader().withIgnoreHeaderCase().withTrim());
+      ObjectMapper objectMapper = new ObjectMapper();
+      for (CSVRecord csvRecord : csvParser) {
+        String orgName = csvRecord.get(PROVIDER_ORGANIZATION_NAME_FIELD);
+        String npi = csvRecord.get(NPI_FIELD);
+        String entityTypeCode = csvRecord.get(ENTITY_TYPE_CODE_FIELD);
+        String taxonomyCode = csvRecord.get(TAXONOMY_CODE_FIELD);
+        String providerFirstName = csvRecord.get(PROVIDER_FIRST_NAME_FIELD);
+        String providerMiddleName = csvRecord.get(PROVIDER_MIDDLE_NAME_FIELD);
+        String providerLastName = csvRecord.get(PROVIDER_LAST_NAME_FIELD);
+        String providerPrefix = csvRecord.get(PROVIDER_PREFIX_FIELD);
+        String providerSuffix = csvRecord.get(PROVIDER_SUFFIX_FIELD);
+        String providerCredential = csvRecord.get(PROVIDER_CREDENTIAL_FIELD);
+        NPIData npiData =
+            NPIData.builder()
+                .npi(npi)
+                .providerOrganizationName(orgName)
+                .entityTypeCode(entityTypeCode)
+                .taxonomyCode(taxonomyCode)
+                .taxonomyDisplay(taxonomyMap.get(taxonomyCode))
+                .providerFirstName(providerFirstName)
+                .providerMiddleName(providerMiddleName)
+                .providerLastName(providerLastName)
+                .providerNamePrefix(providerPrefix)
+                .providerNameSuffix(providerSuffix)
+                .providerCredential(providerCredential)
+                .build();
+        String json = objectMapper.writeValueAsString(npiData);
+        dos.write((npi + "\t" + json).getBytes());
```

Comment:
Since we're in refactoring mode here, I think we way want to also use a CSV writer here instead of hardcoding the tab character. Since we know this NPI data is poorly validated, it's not implausible to think that maybe someone could put a tab character in their org name or something.